js/binance.js

```diff
@@ -2820,7 +2821,7 @@ module.exports = class binance extends Exchange {
         const defaultType = this.safeString2 (this.options, 'createOrder', 'defaultType', 'spot');
         const marketType = this.safeString (params, 'type', defaultType);
         const clientOrderId = this.safeString2 (params, 'newClientOrderId', 'clientOrderId');
-        const postOnly = this.safeValue (params, 'postOnly', false);
+        const postOnly = this.isPostOnly (type.toUpperCase () === 'market', false, params);
```

Comment:
This line looks wrong to me, an uppercase string cannot be equal to lowercase `'market'`, you probably meant this:

```suggestion
        const postOnly = this.isPostOnly (type.toUpperCase () === 'MARKET', false, params);
```

But we don't need to uppercase the type twice, a better solution would be to move line 2852 higher:

```JavaScript
        const initialUppercaseType = type.toUpperCase ();
```